src/is/L42/flyweight/P.java

```diff
@@ -103,10 +117,26 @@ public boolean hasUniqueNum() {
       return false;
     }
 
+    private static record NCsI(int n, List<C> cs) {}
+    private static Map<NCsI, NCs> created = new HashMap<>();
+    public static NCs newNCs(final int n, final List<C> cs) {
+      NCsI ci = new NCsI(n, cs);
+      if(PerfCounters.isEnabled()) {
+        PerfCounters.inc("invoke.P.NCs.init.total");
+        if(!created.containsKey(ci)) {
+          PerfCounters.inc("invoke.P.NCs.init.total.unique");
+          }
+        }
+      return created.computeIfAbsent(ci, ci2->new NCs(ci2.n, ci2.cs));
+      }
+    Object readResolve() throws ObjectStreamException {
+      return newNCs(this.n, this.cs);

```

Comment:
I'm still confused on how this can work: it is a NON static object, so, it must be creating a NCs object, right?
so, is the system creating a NCs object with nulls/default values everywhere and then trashing it to return the result of readResolve?